src/controllers/question.js

```diff
@@ -1,11 +1,98 @@
+// const mongoose = require('mongoose');
+const Question = require('../models/question');
+const Student = require('../models/student');
+
+const filterQuestions = async (res, query) => {
+  try {
+    // Find and sort by latest
+    const questions = await Question.find(query); // .sort({ createdAt: -1 });
+    if (questions.length <= 0) {
+      res.status(200).json({ message: 'No questions found' });
+    } else res.status(200).json(questions);
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+  }
+};
 module.exports = {
-  getAllQuestions: async () => {},
-  getOneQuestion: async () => {},
-  addNewQuestion: async () => {},
+  getAllQuestions: async (req, res) => {
+    /* test if it's registered user then get all the questions
+      related to the interested subjects of the user
+     */
+    const query = {};
+    if (req.user) {
+      // find user's subjects
+      const student = await Student.findById(req.user.id);
+      // if user has no subject then get all questions ordered by date
+      if (!student.subjects) {
+        filterQuestions(res, query);
+      } else {
+        query['subjects.title'] = student.subjects.title;
+        filterQuestions(res, query);
+      }
+    }
+    // if not registered user get all questions ordered by date (recent questions)
+    else {
+      filterQuestions(res, query);
+    }
+  },
+
+  getOneQuestion: async (req, res) => {
+    const { id } = req.params;
+    try {
+      const question = await Question.findById(id);
+      if (question.length <= 0)
```

Comment:
are you receiving a single question or an array of questions here?

Reply:
single question I need to right it like if(!question) right?

Reply:
because question.length is a bit confusing, feels like you're receiving an array

Reply:
yes I fix it